### DISCUSSION TITLE
How do I revert a commit in a GitHub repository?

### DISCUSSION THREAD

**Discussion by mandi-hewage** [community/community#112322] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Could someone explain me the procedure concisely?

**Comment by MostafaWaleed0** [comment]
1. You need to find the commit hash of the commit you want to revert. You can do this by running git log to view the commit history: `git log`
2. Once you have identified the commit you want to revert, use the following command to revert it: `git revert <commit_hash>`
     - This command creates a new commit that undoes the changes made by the specified commit. It's important to note that `git revert` doesn't erase history; instead, it creates a new commit that undoes the changes.

3. After reverting the commit locally, you need to push the changes to the GitHub repository: `git push origin <branch_name>`

**Comment by supunxiii** [comment, marked as answer]
hello! to revert a commit in a GitHub repository all you have to do is,

i. identify the commit to revert: find the commit hash of the commit you want to revert. you can do this by viewing the commit history either on GitHub or locally.

ii. revert the commit using the GitHub interface: on the GitHub repository page, click on the "Commits" tab and find the commit you want to revert. click on the commit to view its details and then click the "Revert" button. this will create a new commit that undoes the changes introduced by the selected commit.

iii. revert the commit locally and push the changes: if you prefer to revert the commit locally you can use git commands. first checkout the branch where the commit exists. then use the command git revert <commit-hash> to create a new commit that undoes the changes. and finally push the changes to the repository.

iv. push the changes: if you reverted the commit locally, don't forget to push the changes to the GitHub repository. 

if there's anything else you want to know and clarify let me know. : )

**Reply by Dave-Allured** [reply]
I tried to follow your procedure (ii) using only the Github interface.  I clicked on the commit to view its details.  I saw the diff, but there was no Revert button.  What do you mean by 'click the "Revert" button"'?

Then I took a hint from your comment.  Using only the Github interface, I used the on-page editor to manually reverse the diff.  That worked, except now that leaves me with two garbage commits in my master branch.  I do not want to push garbage commits to the primary repo.  How do I properly sanitize my master branch, so that it is 100% in sync with the primary, with no garbage commits?

**Reply by PaweX** [reply]
But I want to revert a commit in a proper way, so there is no such option on the website?? I also can't see the revert button.

So if I have a little repo that I don't want to clone to my PC/laptop/fridge whatever, I understand that there is no other way to revert a commit than doing it manually in edit-the-file mode? Great!

**Comment by DilpreetSinghh** [comment]
# Reset local repo to the commit BEFORE `<commit hash>`
git reset --hard `<commit hash>`

# Force push to overwrite GitHub history
git push origin `<branch name>` --force
